lib/ansible/galaxy/collection/galaxy_api_proxy.py

```diff
@@ -52,13 +57,31 @@ def get_collection_versions(self, requirement):
             if isinstance(requirement.src, GalaxyAPI)
             else self._apis
         )
-        return set(
-            (version, api)
-            for api in api_lookup_order
-            for version in api.get_collection_versions(
-                requirement.namespace, requirement.name,
-            )
-        )
+        result = set()
+        last_err = None
+        for api in api_lookup_order:
+            try:
+                for version in api.get_collection_versions(requirement.namespace, requirement.name):
+                    result.add((version, api))
+            except GalaxyError as api_err:
+                last_err = api_err
+            except Exception as err:
+                display.warning(
+                    "Skipping Galaxy server %s. "
+                    "Got an unexpected error when getting "
+                    "available versions of collection %s.%s: %s"
+                    % (
+                        api.api_server,
+                        requirement.namespace,
+                        requirement.name,
+                        to_text(err),
+                    )
+                )
```

Comment:
Have you considered tracking these exceptions too?
```suggestion
                )
                last_err = err
```

Reply:
I did, but wasn't sure. I thought maybe we should show the last legitimate-looking error...

Reply:
Maybe. I don't have a strong opinion here. Maybe to use this exception too we need to count that there were no successful loop iterations (legitimately returning empty results).